### PR TITLE
Add linting to support deprecating component imports

### LINKED ISSUE #20633: Add linting to prevent deprecated component imports
## Issue Description

The Design System team is introducing Web Components. In the ["migration" section of the design doc](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/platform/engineering/design-docs/2021-01-13-web-component-library.md#migration-to-web-components), it was established that we will add an ESLint rule which will throw an error if any deprecated components are imported. In order to avoid excessive violations, this new rule will only run on files that have been changed, and we will manually configure it any time we release a Web Component which replaces an existing React version of a component from `component-library`.

---

## Acceptance Criteria
- [ ] We have linting in place that will throw an error on _new_ imports of deprecated components
---

### PR DESCRIPTION
**Pull request by bkjohnson**
## Description

Closes https://github.com/department-of-veterans-affairs/va.gov-team/issues/20633

As we begin releasing Web Components, we want to be able to prevent teams from adding the legacy React components from our `component-library` to new things that they build. This linting rule will allow us to add a key/value pair anytime we want to throw an error for a specific import.

## Testing done

Local linting

## Screenshots

image

## Acceptance criteria
- [ ]

## Definition of done
- [ ] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [ ] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

File: .eslintrc.js [hunk 1]
```diff
@@ -0,0 +1,246 @@
+module.exports = {
```
Comment (bkjohnson) [inline .eslintrc.js:1]: I copied `.eslintrc`, added `module.exports`, and fixed some of the comments in order to comply with linting (some of them didn't have a space between `//` and the comment string)

File: changed.eslintrc.js [hunk 2]
```diff
@@ -0,0 +1,17 @@
+const eslintConfig = require('./.eslintrc');
```
Comment (bkjohnson) [inline changed.eslintrc.js:1]: Make sure we get the default config

File: changed.eslintrc.js [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+  ...eslintConfig.rules,
+  'deprecate/import': [
+    'error',
+    {
+      name:
+        '@department-of-veterans-affairs/component-library/CollapsiblePanel',
+      use: '<va-accordion>',
+    },
+  ],
+};
```
Comment (bkjohnson) [inline changed.eslintrc.js:15]: Add in the `deprecate/import` rule and configure it to error out on `CollapsiblePanel` imports.

**Comment by bkjohnson** [comment]
Now that https://github.com/department-of-veterans-affairs/vets-website/pull/15941is merged, having this PR merged in will help us prevent the use of older components when there is a Web Component replacement available.

**Comment by alexandec** [comment]
I like this automated approach to handling deprecation! 👍

**Review by alexandec — APPROVED** [review]

**Pull request merged**